oracle/src/main.sw

```diff
@@ -0,0 +1,58 @@
+contract;
+
+dep interface;
+dep data_structures;
+dep errors;
+dep events;
+
+use std::{
+    chain::auth::msg_sender,
+    constants::BASE_ASSET_ID,
+    identity::Identity,
+    logging::log,
+    result::Result,
+    revert::require,
+};
+
+use interface::Oracle;
+use data_structures::State;
+use errors::{AccessError, InitializationError};
+use events::PriceUpdateEvent;
+
+storage {
+    /// The Identity that can control the oracle (node)
+    owner: Identity = Identity::ContractId(BASE_ASSET_ID),
```

Comment:
In order to avoid setting the owner to the `BASE_ASSET_ID`, perhaps set this as an `Option`. There is a section on [Contract Ownership](https://fuellabs.github.io/sway/v0.20.2/blockchain-development/access_control.html#contract-ownership) in the Sway Book.